_posts/2021/2021-08-02-dont-call-replacement-classes-new.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Don't call a new verison of something new"
+description: "A opinionated naming subtlety"
+layout: article
+category: ruby
+image:
+  base: '2020/calculate-a-mean-average-from-a-ruby-array'
+  alt: ""
+  credit: 
+  source: "https://unsplash.com/photos/3xwrg7Vv6Ts"
+
+---
+
+Often in long-lived applications we might want to rewrite or refactor the approach that we have previously taken.
+
+
+## Instead of...
+
+...using 
+
+```ruby
+class NewDoThingJob << ActiveJob
+  # new implementation
+end
+```
+
+or
+
+```ruby
+class DoThingJobVersionTwo << ActiveJob
+  # new implementation
+end
+```
+
+
+## Choose...
+
+to rename or namespace the existing code. And create the replacement version with the correct future naming.
+
+```ruby
+module Removing
+  class DoThingJob << ActiveJob
+    # existing implementation
+  end
+end
+```
+
+```ruby
+class DoThingJob < Removing::DoThingJob
+```
+
+Then change all the places the non-namespaced version is called. Then deploy and ensure that no jobs are still enqueued with the non-namespaced version.
+
+Now you're free to reimplement in the correct (and future) location.
+
+```ruby
+class DoThingJob
+  # new implmentation
+end
+```
+
+Once you've built, tested and changed all the calls to the new version you can delete the (now unused) namespaced version.
+
+
+## Why?
+
+This is a habit that helps communicate intention.
+
+These refactoring projects are often large in scale and easy to make mistakes. But this methodology clearly lays out the intention to replace this existing code.
+
+If you call your new implmentation `NewSomething` it's not easy for other folks in your team (and future you) to understand the direction of development of the code. "Should I use the New version?"
+
+You also won't be left with lots of `NewWhatever` objects littering your code when you inevitably don't do the final renaming.
```

Comment:
Consistency with `NewSomething` earlier?

And maybe suggestion that renaming might not end up happening, at least for a while, instead of the "when you inevitably don't do it..." Unless you wanna keep the sass!